### ISSUE TITLE
Not Working in Android Version 12 and Above

### ISSUE THREAD

**Issue by dev-abhinavsinghHP** [Driversnote-Dev/react-native-kontaktio#121]
it works fine in Android version 11 , for 12 and 13 Beacons did not register beacons are not shown in Range Anyone Please help me with this issue 

I use latest Package

**Comment by KaiValar** [comment]
You need to ask for BLUETOOTH_SCAN and BLUETOOTH_CONNECT permissions too for that newer versions.

**Comment by jkaos92** [comment]
@KaiValar even with BLUETOOTH_SCAN and BLUETOOTH_CONNECT it is not working in Android 14. There is any other advice?

**Comment by KaiValar** [comment]
> @KaiValar even with BLUETOOTH_SCAN and BLUETOOTH_CONNECT it is not working in Android 14. There is any other advice?

Have you a repo to check it out? I have an app running it with this lib on Android 14, but I cannot share the code beacuse isn't really mine, I'm only an employee ^^'

You need to add permissions and the service from Kontakt SDK on Manifest, add the SDK on your build.gradle dependencies (I used 7.1.8 version) and initialize it on MainApplication.java, then you can ask for the permissions to user and finally use the lib. On iOS is more painful, but atm on Android works fine.

**Comment by jkaos92** [comment]
@KaiValar  thanks for the fast reply.
Can't post the whole repo (for the same reason), we followed the guide. It works on iOS for us, but it doesn't work on Android 14 (it works on android 12 tho).

Our Manifest:

```
<uses-permission android:name="android.permission.INTERNET"/>
<uses-permission android:name="android.permission.ACCESS_FINE_LOCATION"/>
<uses-permission android:name="android.permission.ACCESS_NETWORK_STATE"/>
<uses-permission android:name="android.permission.BLUETOOTH" android:maxSdkVersion="30" />
<uses-permission android:name="android.permission.BLUETOOTH_ADMIN" android:maxSdkVersion="30" />
<uses-permission android:name="android.permission.BLUETOOTH_SCAN" />
<uses-permission android:name="android.permission.BLUETOOTH_CONNECT"/>
```

and this: 

```
<service android:name="com.kontakt.sdk.android.ble.service.ProximityService" android:exported="false"/>
```

this in the build.gradle:
```
implementation("io.kontakt.mvn:sdk:7.1.8")`
```

And this is the listener that doesn't work on Android 14 (but works on Android 12). 

```
kontaktEmitter.addListener('beaconsDidUpdate', async (res)=> {
      console.log('beaconsDidUpdate', res);
      //basically "res" is undefined on Android 14
      //it returns the correct array in Android 12 (with beacons and regions) 
  });
```


The initialize onCreate in MainApplication.java is not needed for us because we don't use the API Key (or at least, it doesn't work without an API key).  

Also we followed everything in this guide: https://github.com/Driversnote-Dev/react-native-kontaktio/tree/master


Any recommendation is appreciated. Thanks in advance!

**Comment by jkaos92** [comment]
@KaiValar After many many attempts, we found the solution on a comment of another bluetooth library with similar issue [here](https://github.com/dotintent/react-native-ble-plx/issues/1014#issuecomment-1253789705), where Android 12 worked fine, on Android 14 didn't.

```
<uses-permission android:name="android.permission.BLUETOOTH_SCAN" tools:remove="android:usesPermissionFlags" />
```

Basically this saved the day!


--- 

The problem was that by adding this (mandatory for Android 12+, api 31+):
```
<uses-permission android:name="android.permission.BLUETOOTH_SCAN" />
```
The merged Manifest resulted in this:
```
 <uses-permission android:name="android.permission.BLUETOOTH_SCAN" android:usesPermissionFlags="neverForLocation"/>
```

and by automatically adding this, the Android documentation says:

`Include "neverForLocation" only if you can strongly assert that your app never derives physical location from Bluetooth scan results. If you include neverForLocation in your android:usesPermissionFlags, some BLE beacons are filtered from the scan results.`

Basically the automatically added "neverForLocation" was filtering out some BLE beacons...

**Comment by KaiValar** [comment]
Glad to read you solve that :D

**Comment by KaiValar** [comment]
Btw, @jkaos92 if you are thinking to use this in background, I used a Foreground Service ;)

**Comment by KaiValar** [comment]
@jkaos92 don't use a Foreground Service, is a suicide attempt lol I had to use another approach to hack Android Doze Mode
